### PR TITLE
Add keyword column categories to content categories

### PR DESCRIPTION
**Pull request by mattiasostmar**
Task: https://phabricator.wikimedia.org/T162125

Adds mapped keywords from column <Nyckelord> to content categories.

### CODE REVIEW

File: create_infotexts.py [hunk 1]
```diff
@@ -461,6 +493,16 @@ def process_depicted_place(self, place_string, places_mapping, desc_string):
 
 
         self.data["depicted_place"] = place_as_wikitext
+        
+    
+    def process_keywords(self, item, mapping):
```
Comment (lokal-profil) [inline create_infotexts.py]: An empty function!

Was the plan to have L237-242 inside this function? If so it should be a function under the CypernImage class instead.

File: create_infotexts.py [hunk 2]
```diff
@@ -43,6 +43,31 @@ def load_places_mapping():
...
+    kw1 = kw1[["Commons category", "wikidata"]] # Remove col "fredquency" for boolean filtering to work
+    
+    kw1[kw1 == "-"] = None  # Pandas "boolean indexing" kung-fu to replace all "-" with None
+    
+    kw_dict = {}
+    for index, row in kw1.iterrows():
+        kw_dict[index] = {}
+        if row["Commons category"]:
+            kw_dict[index]["commonscat"] = row["Commons category"]
+        kw_dict[index]["wikidata"] = row["wikidata"] # should always be present
```
Comment (lokal-profil) [inline create_infotexts.py]: Not sure I see the difference between wikidata and commonscat. Looks like neither is guaranteed to be present.

You could add both and then on L240 use `if keywords_mapping[kw]["commonscat"]:` to catch the `none` cases

File: create_infotexts.py [hunk 3]
```diff
@@ -471,15 +513,14 @@ def generate_list_of_stripped_keywords(self, keyword_string):
         :return: list of keywords.
         """
         keywords_list = keyword_string.split(", ")
-        keywords_list_low = [kw.lower() for kw in keywords_list]
```
Comment (lokal-profil) [inline create_infotexts.py:474]: Why the change? Wasn't the reason for converting these that the keywords in the metadata were a mixture of upper and lower case?

File: create_infotexts.py [hunk 4]
```diff
@@ -43,6 +43,31 @@ def load_places_mapping():
...
+    Read wikitable html into Pandas DataFrame, transform it and return a dictionary.
+    
+    :return: dictionary
+    """
+    kw_maps_url = "https://commons.wikimedia.org/wiki/Commons:Medelhavsmuseet/batchUploads/Cypern_keywords"
+    keywords = pd.read_html(kw_maps_url, attrs={"class": "wikitable sortable"}, header=0)
+    kw1 = keywords[0] # First is the actual column <Nyceklord>
+    
+    kw1 = kw1.set_index("Nyckelord")
+    kw1 = kw1[["Commons category", "wikidata"]] # Remove col "fredquency" for boolean filtering to work
```
Comment (lokal-profil) [inline create_infotexts.py]: typo in comment

**Review by lokal-profil — CHANGES_REQUESTED** [review]

File: create_infotexts.py (hunk 4 above)
Comment (mattiasostmar) [reply create_infotexts.py]: Fixed

**Comment by lokal-profil** [comment]
Does this harvest all three tables on the wiki page? If so is it possible to just harvest the first?

File: create_infotexts.py (hunk 2 above)
Comment (mattiasostmar) [reply create_infotexts.py]: Yes, a check on L240 is better. It complained if I used that syntax though, so used .get() syntax. Done.

File: create_infotexts.py (hunk 1 above)
Comment (mattiasostmar) [reply create_infotexts.py]: Ah yes, I figured out the logic directly in L237-L242 and forgot to add it as a object method. Done!

File: create_infotexts.py (hunk 3 above)
Comment (mattiasostmar) [reply create_infotexts.py:474]: Actually, when I looked at the original metadata again I noticed that it was all lower case. Thus, I reduced complexity by removing this and gives the same end result.

**Comment by mattiasostmar** [comment]
Yes it harvests all three tables on the wiki page, but L54 selects only the first table by slicing.

 The syntax "kw1" is a left-over from my original plan to use all three (which was a bad idea). I rename "kw1" to "keywords" now to reduce the risk of confusion.

Done.

File: create_infotexts.py (hunk 2 above)
Comment (lokal-profil) [reply create_infotexts.py]: Ah that is because you are still doing
```python
if row["commonscat"]:
    places_dict[index]["commonscat"] = row["commonscat"]
```
on L39
If you change that to ```places_dict[index]["commonscat"] = row["commonscat"]``` then you don't need the `.get()`.

But it's just about consistency so not a blocker.

File: create_infotexts.py [hunk 5]
```diff
@@ -495,14 +492,19 @@ def process_depicted_place(self, place_string, places_mapping, desc_string):
...
         
     
-    def process_keywords(self, item, mapping):
+    def process_keywords(self, mapping):
         """
         Add potential content categories from column <Nyckelord>.
         
         Populates self.content_cats
         
         :return: None
```
Comment (lokal-profil) [inline create_infotexts.py]: No need to mention the `:return: None` (since you mention that it populates self.content_cats)

File: create_infotexts.py (hunk 3 above)
Comment (lokal-profil) [reply create_infotexts.py:474]: OK

File: create_infotexts.py [hunk 6]
```diff
@@ -50,16 +50,16 @@ def load_keywords_mapping():
...
     kw_maps_url = "https://commons.wikimedia.org/wiki/Commons:Medelhavsmuseet/batchUploads/Cypern_keywords"
-    keywords = pd.read_html(kw_maps_url, attrs={"class": "wikitable sortable"}, header=0)
-    kw1 = keywords[0] # First is the actual column <Nyceklord>
-    
-    kw1 = kw1.set_index("Nyckelord")
-    kw1 = kw1[["Commons category", "wikidata"]] # Remove col "fredquency" for boolean filtering to work
-    
-    kw1[kw1 == "-"] = None  # Pandas "boolean indexing" kung-fu to replace all "-" with None
+    tables = pd.read_html(kw_maps_url, attrs={"class": "wikitable sortable"}, header=0)
+    keywords = tables[0] # First table is the actual column <Nyceklord>
```
Comment (lokal-profil) [inline create_infotexts.py]: Formatting for comments should be `<code>  # <comment>` i.e. two spaces before and one after (also true for some other comments).

Could possibly also clarify the comment by changing it to `# First table is the one corresponding to <Nyckelord>` since column becomes a bit confusing when we are talking about tables (also fixed typo).

**Review by lokal-profil — APPROVED** [review]
I left a few comments about minor cosmetic changes/clarifications but none which should be considered blocking. I.e. feel free to fix them or not.

The renaming made it a lot clearer that we are only looking at the first table. Thanks =)

File: create_infotexts.py (hunk 6 above)
Comment (mattiasostmar) [reply create_infotexts.py]: Good, done!
